File: JornadaLaboral/Métodos/JornadaLaboral.java
package Métodos;

public class JornadaLaboral {
    private int horasTrabajadas;
    private int horasExtras;
    private double salario;
    private double total;
    private double totalHorasExtras;
    private double salarioTotal;

    //MétodoConstructor
    public JornadaLaboral (int horasTrabajadas, int salario){
        this.horasTrabajadas = horasTrabajadas;
        this.salario = salario;
        this.calcularSalarioNormal();
    }
    //Constructor 2
    //Sobrecarga
    public JornadaLaboral(int horasTrabajadas, int horasExtras, double salario) {
        this.horasTrabajadas = horasTrabajadas;
        this.horasExtras = horasExtras;
        this.salario = salario;
        this.calcularCostoHorasExtras();
        this.calcularSalarioNormal();
        this.calcularSalarioSemanal();
    }

    //get
    public double getTotal() { return total; }

    public double getSalarioTotal() {
        return salarioTotal;
    }


    //Método de instancia

    public void calcularSalarioNormal() {
        this.total = this.horasTrabajadas * this.salario;
    }

    public void calcularCostoHorasExtras() {
        this.horasExtras = horasExtras;
        this.salario = salario;
        this.totalHorasExtras += this.horasExtras * (this.salario * 1.5);
    }

    public void calcularSalarioSemanal() {
        this.salarioTotal = this.total + this.totalHorasExtras;
    }


    //Método de clase
    public static double calcularSalarioNormal(double horasTrabajadas, double salario) {
        double total = horasTrabajadas * salario;
        return total;
    }
}

File: JornadaLaboralP/Métodos/JornadaLaboral.py
class JornadaLaboral:

    __horasTrabajadas = int(0)
    __horasExtras = int(0)
    __salario = float(0.0)
    __total = float(0.0)
    __totalHorasExtras = float(0.0)
    __salarioTotal = float(0.0)

    #Método constructor
    def __init__(self, horasTrabajadas, horasExtras, salario):
        self.__horasTrabajadas = horasTrabajadas
        self.__horasExtras = horasExtras
        self.__salario = salario
        self.calcularSalarioNormal()
        self.calcularCostoHorasExtras()
        self.calcularSalarioSemanal()

    def get_SalarioTotal(self):
        return self.__salarioTotal

    #Método destructor
    def __del__(self):
        self.calcularSalarioNormal()
        self.calcularCostoHorasExtras()
        self.calcularSalarioSemanal()

    #Método de instancia
    def calcularSalarioNormal(self):
        self.__total = self.__horasTrabajadas * self.__salario

    def calcularCostoHorasExtras(self):
        self.__totalHorasExtras = self.__horasExtras * (self.__salario * 1.5)

    def calcularSalarioSemanal(self):
        self.__salarioTotal = self.__total + self.__totalHorasExtras

    #Método de clase
    @classmethod
    def calcularSalario (cls, horasTrabajadas, salario):
        return horasTrabajadas*salario

    #Método de instancia
    def calcularSalarioNormal(self):
        self.__total = self.__horasTrabajadas * self.__salario
    def get_salarioNormal(self):
        return self.__total

